File: scratch/check_headers.py
import os
import re

pages_dir = r"d:\web_develop_project\QAISFOODS\src\pages"
files = [f for f in os.listdir(pages_dir) if f.endswith(".tsx")]

for file in files:
    path = os.path.join(pages_dir, file)
    with open(path, "r", encoding="utf-8") as f:
        content = f.read()
    
    # search for <h1> or similar that has text-2xl font-bold and a <p> tag right after
    # Let's print out the matches
    matches = re.finditer(r'(<h1.*?/h1>\s*(?:{.*?}\s*)*<p className=".*text-muted-foreground.*">.*?</p>)', content, re.DOTALL | re.IGNORECASE)
    for m in matches:
        print(f"File: {file}")
        print(m.group(0))
        print("-" * 40)

File: scratch/check_headers.js
import fs from 'fs';
import path from 'path';

const pagesDir = 'd:\\web_develop_project\\QAISFOODS\\src\\pages';
const files = fs.readdirSync(pagesDir).filter(f => f.endsWith('.tsx'));

const results = [];

files.forEach(file => {
  const filePath = path.join(pagesDir, file);
  const content = fs.readFileSync(filePath, 'utf8');
  
  const lines = content.split('\n');
  lines.forEach((line, idx) => {
    if (line.includes('<h1') || line.includes('font-bold">')) {
      const isHeader = line.includes('text-2xl') || line.includes('text-3xl') || line.includes('text-4xl') || line.includes('<h1');
      if (isHeader) {
        // Look for the next few lines for any <p> tag with text-muted-foreground
        let foundP = null;
        for (let i = idx + 1; i < Math.min(lines.length, idx + 6); i++) {
          if (lines[i].includes('<p') && (lines[i].includes('text-muted-foreground') || lines[i].includes('text-sm'))) {
            foundP = { lineNum: i + 1, content: lines[i].trim() };
            break;
          }
        }
        results.push({
          file,
          headerLine: idx + 1,
          headerContent: line.trim(),
          description: foundP
        });
      }
    }
  });
});

console.log(JSON.stringify(results, null, 2));
